scheduler/src/cook/kubernetes/controller.clj

```diff
@@ -160,41 +188,112 @@
+                                   (weird expected-state-dict pod)
+                                   (pod-was-killed kcc pod-name))
+                                 ; These cases are weird.
+                                 [:expected/completed :pod/waiting] (kill-task-weird api-client expected-state-dict pod)
+                                 [:expected/completed :pod/running] (kill-task-weird api-client expected-state-dict pod)
+
+                                 ; This is interesting. This indicates that something deleted it behind our back!
+                                 ; Other 3 cases of [{completed,starting,killed}, :missing] already handled.
+                                 [:expected/running :missing] (do
+                                                                (log/error "Something deleted" pod-name "behind our back")
+                                                                (pod-has-just-completed kcc existing-state-dict)) ; TODO: Should mark mea culpa retry
+
+                                 ; This block of :missing,* cases can only occur in testing when you're e.g., blowing away the database. 
+                                 ; The next two will go through :missing,running or :missing,waiting to :missing,failed and then be deleted.
+                                 ; TODO: These may be evidence of a bug where we process pod changes when we're starting up.
+                                 [:missing :pod/running] (kill-task-weird api-client nil pod)
+                                 [:missing :pod/waiting] (kill-task-weird api-client nil pod)
+
+                                 ; We shouldn't hit these unless we get a database rollback.
+                                 [:missing :pod/succeeded] (kill-task-weird api-client nil pod)
+                                 [:missing :pod/failed] (kill-task-weird api-client nil pod)
+
+                                 [:missing :pod/unknown] (kill-task-weird api-client expected-state-dict pod)
+                                 [:expected/starting :pod/unknown] ; TODO: Should mark mea culpa retry
+                                 (kill-task-weird api-client (pod-has-just-completed kcc existing-state-dict) pod)
+                                 [:expected/running :pod/unknown] ; TODO: Should mark mea culpa retry
+                                 (kill-task-weird api-client (pod-has-just-completed kcc existing-state-dict) pod)
+                                 [:expected/killed :pod/unknown] ; TODO: Should mark mea culpa retry
+                                 (kill-task-weird api-client (pod-has-just-completed kcc existing-state-dict) pod)
+                                 [:expected/completed :pod/unknown] ; TODO: Should mark mea culpa retry
+                                 (kill-task-weird api-client (pod-has-just-completed kcc existing-state-dict) pod)
```

Comment:
Nitpick: `compute-cluster` instead of `kcc`?

Reply:
It was more confusing to me since `kcc` is only used in this package and outside every where else it is called `compute-cluster`. I don't see the compute cluster being referred to as `mcc` in the `mesos` package.

Reply:
Ok.

Reply:
Has this been committed?